tests/test_experiment.py

```diff
@@ -19,11 +19,7 @@ def klass(self):
         return Experiment
 
     @pytest.fixture
-    def exp(self, klass):
```

Comment:
As above, I think we may still need something like this for experiment configuration. Though something explicit like `exp(setup=False)` or similar would probably be better than implicitly treating the `session` as a flag.

Reply:
Missed this comment while replying to the previous one, but I think we're aligned here. Not passing a session is a weird way to signal you don't want to run setup.